community-artifacts/madlib_image_loader.py

```
@@ -0,0 +1,384 @@
+            self._copy_into_db(f, data)
+
+    def load_np_array_to_table(self, data_x, data_y, table_name, append=False,
+                               img_names=None, no_temp_files=False):
+        """
+        Loads a numpy array into db.  For append=False, creates a new table and loads the data.
+            For append=True, appends data to existing table.  Throws an exception if
+            append=False and table_name already exists, or if append=True and table_name
+            does not exist.  Makes use of worker processes initialized during ImageLoader
+            object creation to load in parallel.
+        @data_x independent variable data, a numpy array of images.  Size of first dimension is
+                number of images.  Rest of dimensions determined by image resolution and number
+                of channels.
+        @data_y dependent variable data (image classes), as an numpy array
+        @table_name Name of table in db to load data into
+        @append Whether to create a new table (False) or append to an existing one (True).
+                If unspecified, default is False
+        @img_names If not None, a list of the image names corresponding to elements of the
+                   data_x numpy array.  If present, this is included as a column in the table.
+        @no_temp_files If specified, no temporary files are written--all operations are performed
+                       in-memory.
+
+        """
+        start_time = time.time()
+        self.mother = True
+        self.append = append
+        self.table_name = table_name
+        self.db_connect()
+
+        if self.append:
+            # Validate that table already exists
+            try:
+                self.db_exec("SELECT count(*) FROM {0}".format(self.table_name), echo=False)
+            except(Exception, db.DatabaseError):
+                raise exceptions.RuntimeError("Table {0} does not exist in {1} db.  Use append=False to create it before loading."
```

We might want to raise the actual error message as well, in case the issue is not table not existing. For example, the user might not have access to the table.